driver-core/src/main/com/mongodb/internal/operation/WriteConcernHelper.java

```diff
@@ -61,9 +61,12 @@ public static boolean hasWriteConcernError(final BsonDocument result) {
     }
 
     public static MongoWriteConcernException createWriteConcernException(final BsonDocument result, final ServerAddress serverAddress) {
-        return new MongoWriteConcernException(
+        MongoWriteConcernException writeConcernException = new MongoWriteConcernException(
                 createWriteConcernError(result.getDocument("writeConcernError")),
                 WriteConcernResult.acknowledged(0, false, null), serverAddress);
+        result.getArray("errorLabels", new BsonArray()).stream().map(i -> i.asString().getValue())
+                .forEach(writeConcernException::addLabel);
+        return writeConcernException;
```

Comment:
It would be natural to consolidate the code in the two methods changed in this PR, but the best place for that it probably in MongoWriteConcernException, and that would require an API changes.  But I want to get this fix into a patch release so am avoiding API changes.  

I think we could do the refactoring as part of a larger consideration of how write concern error are handled, but let me know your thoughts on it.